src/clusterfuzz/_internal/bot/testcase_manager.py

```diff
@@ -1113,12 +1113,10 @@ def setup_user_profile_directory_if_needed(user_profile_directory):
 
 
 def check_for_bad_build(job_type, crash_revision):
-  """Returns result of bad build check i.e. crashes on startup. Results include
-  whether the build is bad, whether the crash should be ignored, the
-  build_run_console_output."""
+  """Return true if the build is bad, i.e. crashes on startup."""
   # Check the bad build check flag to see if we want do this.
   if not environment.get_value('BAD_BUILD_CHECK'):
-    return False, False, None
```

Comment:
I think the issue was possibly coming from here, we should return empty string `''` instead of `None`. That was the default value as you can see on line 1142

Reply:
I think your suggestion fixes the immediate problem, but I'm not so sure it's a proper fix. I feel like update_build_metadata should not be called at all in this case.
Let's reland this tomorrow. I have to go offline so I'll revert rather than leave it broken. I can reland first thing in the

Reply:
You are right, we need a way to skip calling `update_build_metadata` when `BAD_BUILD_CHECK` is not set and `check_for_bad_build` returns early.